From 5581ba196d826a984fbfaf792b7d58535f9911ce Mon Sep 17 00:00:00 2001
Message-ID: <5581ba196d826a984fbfaf792b7d58535f9911ce.1791178679.git.yann.morin@orange.com>
From: "Gary E. Miller" <gem@rellim.com>
Date: Wed, 1 Jul 2026 17:55:57 -0700
Subject: [PATCH] clients/gpsprof.py.in: Quote double quotes in title.

Someone could use the double quote to break out of the
string and add gnuplot commnds.

For issue 404.
Reported by: CuB3y0nd, and Wade Sparks <wsparks@vulncheck.com>
Upstream: https://gitlab.com/gpsd/gpsd/-/commit/5581ba196d826a984fbfaf792b7d58535f9911ce
CVE: CVE-2026-58459
Signed-off-by: Yann E. MORIN <yann.morin@orange.com>

---
 clients/gpsprof.py.in | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/clients/gpsprof.py.in b/clients/gpsprof.py.in
index 5c18f50ff..261e72665 100644
--- a/clients/gpsprof.py.in
+++ b/clients/gpsprof.py.in
@@ -198,6 +198,10 @@ class plotter(object):
         if 'subtype' in self.device:
             desc += "\\n%s" % self.device['subtype']
 
+        # escape ", and \n, for gnuplot, to not break strings
+        desc = desc.replace('"', '\\042')
+        desc = desc.replace('\n', '')
+
         return desc
 
     def collect(self, verb, log_fp=None):
@@ -1262,10 +1266,10 @@ if __name__ == '__main__':
         # Ship the plot to standard output
         if not options.title:
             options.title = plot.whatami()
-            # escape " for gnuplot
-            options.title = options.title.replace('"', '\\"')
         if options.subtitle:
             options.title += '\\n' + options.subtitle
+        # escape " for gnuplot, to not break strings
+        options.title = options.title.replace('"', '\\042')
         term_opts = ""
         truecolor_terms = ['png', 'sixelgd', 'wxt']
         if options.terminal in truecolor_terms:
-- 
2.43.0

